scraper/pipelines.py

```diff
@@ -1,38 +1,51 @@
-import hashlib
-import os
+from hashlib import sha1
+from pathlib import Path
+from urllib.parse import urlparse
 
-from scraper.settings import FILES_STORE, KEEP_FILES
+from scraper.settings import ASYNC_FILE_PROCESSING, FILES_STORE, KEEP_FILES
 from scrapy.pipelines.files import FilesPipeline
 from scrapy.utils.python import to_bytes
-from six.moves.urllib.parse import urlparse
-from tika import parser
+
+from datasets.tasks import content_from_file
 
 
 class ExtractFileContentPipeline(FilesPipeline):
     def file_path(self, request, response=None, info=None):
-        """Retorna onde o arquivo foi baixado.
-
-        Copiado de https://github.com/okfn-brasil/diario-oficial/
-        Issue no scrapy: https://github.com/scrapy/scrapy/issues/4225
+        """Retorna um nome único para o arquivo foi baixado, removendo
+        parâmetros da URL. Por exemplo:
 
         de:
         8e61990b27c6158edaaa715ea76eca65459d92f4.asp?cat=PMFS&dt=03-2016
         para:
         8e61990b27c6158edaaa715ea76eca65459d92f4.asp
+
+
+        Issue no scrapy: https://github.com/scrapy/scrapy/issues/4225
         """
-        url = request.url
-        media_guid = hashlib.sha1(to_bytes(url)).hexdigest()
-        media_ext = os.path.splitext(url)[1]
-        if not media_ext.isalnum():
-            media_ext = os.path.splitext(urlparse(url).path)[1]
-        return "full/%s%s" % (media_guid, media_ext)
+        uid = sha1(to_bytes(request.url)).hexdigest()
+        extension = Path(urlparse(request.url).path).suffix
+        return f"full/{uid}{extension}"
 
     def item_completed(self, results, item, info):
-        if results and results[0][0]:
-            file_info = results[0][1]
-            file_path = f"{FILES_STORE}{file_info['path']}"
-            raw = parser.from_file(file_path)
-            item["file_content"] = raw["content"]
-            if KEEP_FILES is False:
-                os.remove(file_path)
-        return item
+        if not results:
+            return
+
+        for result in results:
+            ok, file_info = result
+            if not ok:
+                continue
+
+            kwargs = {
+                "item_name": item.__name__,
+                "url": file_info["url"],
+                "path": f"{FILES_STORE}{file_info['path']}",
+                "checkusm": file_info["checksum"],
```

Comment:
```suggestion
                "checksum": file_info["checksum"],
```
Typo honesto mas causa um erro na execução da task. :sweat_smile: